crates/bevy_math/src/primitives/mod.rs

```diff
@@ -13,6 +13,26 @@ pub trait Primitive2d {}
 /// A marker trait for 3D primitives
 pub trait Primitive3d {}
 
+/// An error indicating that a direction is invalid.
+#[derive(Debug, PartialEq)]
+pub enum InvalidDirectionError {
+    /// The length of the direction vector is zero or very close to zero.
+    Zero,
+    /// The length of the direction vector is `std::f32::INFINITY`.
+    Infinite,
+    /// The length of the direction vector is `NaN`.
+    NaN,
+}
+
+impl std::fmt::Display for InvalidDirectionError {
```

Comment:
Is there any reason why `thiserror` is not used here?

Reply:
`bevy_math` doesn't have `thiserror` as a dependency currently. Could add it though, especially if we add more error types